### PR TITLE
Adding old token checks

### PR DESCRIPTION
**Pull request by secondsun**

### CODE REVIEW

**Comment by secondsun** [comment]
@matzew @danielpassos :eyeglasses: ?

File: aerogear-android-push/src/main/java/org/jboss/aerogear/android/unifiedpush/fcm/UnifiedPushInstanceIDListenerService.java [hunk 1]
```diff
@@ -85,11 +85,21 @@ public void onTokenRefresh() {
                 try {
                     String token = instanceID.getToken();
                     JsonObject oldPostData = new JsonParser().parse(preference.getValue().toString()).getAsJsonObject();
+                    String oldToken = "";
+                    try {
+                        oldToken = oldPostData.get("deviceToken").getAsString();
+                    } catch (Exception exception) {
+                        //We don't care if the old device token isn't set right.
+                        Log.w(TAG, exception.getMessage(), exception);
```
Comment (edewit) [inline aerogear-android-push/src/main/java/org/jboss/aerogear/android/unifiedpush/fcm/UnifiedPushInstanceIDListenerService.java]: is warning really the right level here? if you update the lib you'll always hit this error, right?

**Comment by edewit** [comment]
@secondsun @matzew could you link the https://github.com/aerogear/aerogear-unifiedpush-server/ PR that corresponds to this one.

File: aerogear-android-push/src/main/java/org/jboss/aerogear/android/unifiedpush/fcm/UnifiedPushInstanceIDListenerService.java (hunk 1 above)
Comment (secondsun) [inline aerogear-android-push/src/main/java/org/jboss/aerogear/android/unifiedpush/fcm/UnifiedPushInstanceIDListenerService.java]: No, we currently store this information.  If the warning is hit then that means that somehow the token value got corrupted or wasn't stored correctly.

**Comment by danielpassos** [comment]
I have tested it and it works like a charm. @matzew Do you wanna do some before I merge?

**Comment by edewit** [comment]
:+1:

**Pull request merged**